app/frontend/src/features/auth/signup/Signup.test.tsx

```diff
@@ -0,0 +1,330 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { QUESTIONS_OPTIONAL } from "components/ContributorForm/constants";
+import { EditLocationMapProps } from "components/EditLocationMap";
+import {
+  EMAIL_LABEL,
+  FEMALE,
+  HOSTING_STATUS,
+  NAME_LABEL,
+  SIGN_UP,
+  SIGN_UP_AWAITING_EMAIL,
+  SIGN_UP_BIRTHDAY,
+  SIGN_UP_REDIRECT,
+  SIGN_UP_TOS_ACCEPT,
+  USERNAME,
+} from "features/auth/constants";
+import { SUBMIT } from "features/constants";
+import { hostingStatusLabels } from "features/profile/constants";
+import { StatusCode } from "grpc-web";
+import { HostingStatus } from "proto/api_pb";
+import { SignupFlowRes } from "proto/auth_pb";
+import { Route, Switch } from "react-router-dom";
+import { signupRoute } from "routes";
+import { service } from "service";
+import { getHookWrapperWithClient } from "test/hookWrapper";
+import { assertErrorAlert, mockConsoleError, MockedService } from "test/utils";
+
+import Signup from "./Signup";
+
+const startSignupMock = service.auth.startSignup as MockedService<
+  typeof service.auth.startSignup
+>;
+const signupFlowAccountMock = service.auth.signupFlowAccount as MockedService<
+  typeof service.auth.signupFlowAccount
+>;
+const signupFlowFeedbackMock = service.auth.signupFlowFeedback as MockedService<
+  typeof service.auth.signupFlowFeedback
+>;
+const validateUsernameMock = service.auth.validateUsername as MockedService<
+  typeof service.auth.validateUsername
+>;
+
+const View = () => {
+  return (
+    <Switch>
+      <Route path={signupRoute}>
+        <Signup />
+      </Route>
+      <Route>
+        <p data-testid="dashboard">Dashboard</p>
+      </Route>
+    </Switch>
+  );
+};
+
+jest.mock("components/EditLocationMap", () => ({
+  __esModule: true,
+  default: (props: EditLocationMapProps) => (
+    <input
+      data-testid="edit-location-map"
+      onChange={(event) => {
+        props.updateLocation({
+          lat: 1,
+          lng: 2,
+          address: event.target.value,
+          radius: 5,
+        });
+      }}
+    />
+  ),
+}));
+
```

Comment:
There are two missing cases here:
- When the component is rendered in a route with a `urlToken`
- When using the `urlToken` to continue the sign up throws an error

The code coverage report has actually been pretty useful for spotting these cases in this instance, as I usually don't obsess over 100% coverage if they are low value things that's gonna be a pain to test. But they seem reasonable to do here and important use cases!